test/src/tests-identity-utils.ts

```diff
@@ -0,0 +1,452 @@
+                'identify:device_application_stamp=test-device-id;customerid=id1;';
+
+            expect(updatedMpIdCache!.hasOwnProperty(cachedKey)).to.equal(true);
+
+            const cachedIdentityCall: ICachedIdentityCall = updatedMpIdCache![cachedKey];
+
+            expect(cachedIdentityCall).hasOwnProperty('responseText');
+            expect(cachedIdentityCall).hasOwnProperty('status');
+            expect(cachedIdentityCall).hasOwnProperty('expireTimestamp');
+
+            expect(cachedIdentityCall.status).to.equal(200);
+            expect(cachedIdentityCall.expireTimestamp).to.equal(currentTime);
+
+            const responseText = JSON.parse(cachedIdentityCall.responseText);
+            expect(responseText).to.deep.equal(identifyResponse);
+        });
+
+        it('should save a login request to local storage', () => {
+            const mpIdCache = window.localStorage.getItem(localStorageIDKey);
+            expect(mpIdCache).to.equal(null);
+            
+            const loginResponse = {
+                ...identifyResponse,
+                is_logged_in: true,
+            };
+            
+
+            const jsonString = JSON.stringify(loginResponse);
+
+            const xhr: XMLHttpRequest = {
+                status: 200,
+                responseText: jsonString,
+            } as XMLHttpRequest;
+
+            const currentTime = new Date().getTime();
+
+            cacheIdentityRequest(
+                Login,
+                knownIdentities,
+                currentTime,
+                cacheVault,
+                xhr
+            );
+
+            const updatedMpIdCache = cacheVault.retrieve();
+
+            expect(Object.keys(updatedMpIdCache!).length).to.equal(1);
+            const cachedKey = 'login:device_application_stamp=test-device-id;customerid=id1;';
+            expect(updatedMpIdCache!.hasOwnProperty(cachedKey)).to.equal(true);
+
+            const cachedLoginCall: ICachedIdentityCall = updatedMpIdCache![
+                cachedKey
+            ];
+
+            expect(cachedLoginCall).hasOwnProperty('responseText');
+            expect(cachedLoginCall).hasOwnProperty('status');
+            expect(cachedLoginCall).hasOwnProperty('expireTimestamp');
+
+            expect(cachedLoginCall.status).to.equal(200);
+            expect(cachedLoginCall.expireTimestamp).to.equal(currentTime);
+
+            const responseText = JSON.parse(cachedLoginCall.responseText);
+            expect(responseText).to.deep.equal(loginResponse);
+        });
+    });
+
+    describe('#concatenateIdentities', () => {
+        it('should return a concatenated user identity string based on the order of IdentityType enum', () => {
+            const userIdentities: IKnownIdentities = {
+                device_application_stamp: 'first',
+                customerid: '01',
+                email: '07',
+                other: '00',
+                other2: '10',
+                other3: '11',
+                other4: '12',
```

Comment:
change the values from stringified numbers to more human readable strings because the numbers could be confusing given the method uses array indices.  The two could be conflated.